scripts/visualizer-polygon.py

```diff
@@ -0,0 +1,271 @@
+# This python file takes argv[1] as path to polygon file to visualization
+# Polygon file format is as same as test/polygons
+
+import matplotlib.pyplot as plt
+import numpy as np
+import sys
+from typing import List, Dict, Any
+from matplotlib.path import Path
+from matplotlib.patches import PathPatch
+
+def patchify(polys):
+    """Returns a matplotlib patch representing the polygon with holes.
+    
+    polys is an iterable (i.e list) of polygons, each polygon is a numpy array 
+    of shape (2, N), where N is the number of points in each polygon.
+    The first polygon is assumed to be the exterior polygon and the rest are holes.
+    The first and last points of each polygon may or may not be the same.
+    
+    This is inspired by https://sgillies.net/2010/04/06/painting-punctured-polygons-with-matplotlib.html
+    """
+    
+    def reorder(poly, cw=True):
+        """Reorders the polygon to run clockwise or counter-clockwise according to the value of cw.
+        It calculates whether a polygon is cw or ccw by summing (x2-x1)*(y2+y1) for all edges 
+        of the polygon, see https://stackoverflow.com/a/1165943/898213.
+        """
+        # Close polygon if not closed
+        if not np.allclose(poly[:, 0], poly[:, -1]):
+            poly = np.c_[poly, poly[:, 0]]
+        
+        direction = ((poly[0] - np.roll(poly[0], 1)) * (poly[1] + np.roll(poly[1], 1))).sum() < 0
+        if direction == cw:
+            return poly
+        else:
+            return poly[:, ::-1]
+    
+    def ring_coding(n):
+        """Returns a list of len(n) of this format: 
+        [MOVETO, LINETO, LINETO, ..., LINETO, LINETO, CLOSEPOLY]
+        """
+        codes = [Path.LINETO] * n
+        codes[0] = Path.MOVETO
+        codes[-1] = Path.CLOSEPOLY
+        return codes
+    
+    # First polygon (exterior) should be CCW, holes should be CW
```

Comment:
I don't think we assume this - perhaps they should be sorted based on containment?